src/nn/webgl/WebGLModel.js

```diff
@@ -0,0 +1,328 @@
+          filterW = operands[inputs[i++]].value[0];
+          filterH = operands[inputs[i++]].value[0];
+          activation = FuseFunctionMap.get(operands[inputs[i++]].value[0]);
+          if (op === OperationCode.AVERAGE_POOL_2D) {
+            output.assign(activation(
+              input.avgPool([filterH, filterW], [strideH, strideW], padding)));
+          } else {
+            output.assign(activation(
+              input.maxPool([filterH, filterW], [strideH, strideW], padding)));
+          }
+        } else {
+          let paddingLeft = operands[inputs[i++]].value[0];
+          let paddingRight = operands[inputs[i++]].value[0];
+          let paddingTop = operands[inputs[i++]].value[0];
+          let paddingBottom = operands[inputs[i++]].value[0];
+          strideW = operands[inputs[i++]].value[0];
+          strideH = operands[inputs[i++]].value[0];
+          filterW = operands[inputs[i++]].value[0];
+          filterH = operands[inputs[i++]].value[0];
+          activation = FuseFunctionMap.get(operands[inputs[i++]].value[0]);
+          if (op === OperationCode.AVERAGE_POOL_2D) {
+            output.assign(activation(
+              input.pad([[0, 0], [paddingTop, paddingBottom], [paddingLeft, paddingRight], [0, 0]])
+                   .avgPool([filterH, filterW], [strideH, strideW], 'valid')));
+          } else {
+            output.assign(activation(
+              input.pad([[0, 0], [paddingTop, paddingBottom], [paddingLeft, paddingRight], [0, 0]])
+                   .maxPool([filterH, filterW], [strideH, strideW], 'valid')));
+          }
+        }
+      } break;
+      case OperationCode.SOFTMAX: {
+        let input = operands[inputs[0]];
+        let beta = operands[inputs[1]].value[0];
+        let output = operands[outputs[0]];
+        output.assign(input.mul(tf.scalar(beta)).softmax());
+      } break;
+      case OperationCode.RESHAPE: {
+        let input = operands[inputs[0]];
+        let targetShape = operands[inputs[1]];
+        let output = operands[outputs[0]];
+        output.assign(input.reshape(targetShape.dataSync()));
+      } break;
+      case OperationCode.CONCATENATION: {
+        if (outputs.length < 1 || inputs.length < 2) {
+          throw new Error('Invalid inputs or outputs');
+        }
+        let numInputTensors = inputs.length - 1;
+        let axis = operands[inputs[numInputTensors]].value[0];
+        let output = operands[outputs[0]];
+        let inputTensors = [];
+        for (let i = 0; i < numInputTensors; ++i) {
+          inputTensors.push(operands[inputs[i]]);
+        }
+        output.assign(tf.concat(inputTensors, axis));
+      } break;
+      case OperationCode.FULLY_CONNECTED: {
+        let input = operands[inputs[0]];
+        let weights = operands[inputs[1]];
+        let bias = operands[inputs[2]];
+        let activation = FuseFunctionMap.get(operands[inputs[3]].value[0]);
+        let output = operands[outputs[0]];
+        let batchSize = input.shape[0];
```

Comment:
`batchSize` should be `product (input.shape) / weights.shape[1]`, according to android nn API spec: https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1ggaabbe492c60331b13038e39d4207940e0aaada7a3dbaf4676aba560c933ff610c5